test/unit/test_logger.py

```diff
@@ -126,3 +126,33 @@ def test_cyan_text():
     x = '{}{}{}'.format(colorama.Fore.CYAN, 'foo', colorama.Style.RESET_ALL)
 
     assert x == logger.cyan_text('foo')
+
+
+def test_markup_detection_pycolors0(monkeypatch):
+    monkeypatch.setenv('PY_COLORS', '0')
+    assert not logger.should_do_markup()
+
+
+def test_markup_detection_pycolors1(monkeypatch):
+    monkeypatch.setenv('PY_COLORS', '1')
+    assert logger.should_do_markup()
+
+
+class FileHandleTTY(object):
+
+    def __init__(self, tty=True):
+        self._tty = tty
+
+    def isatty(self):
+        return self._tty
+
+
+def test_markup_detection_tty_yes(monkeypatch):
+    monkeypatch.delenv('PY_COLORS', raising=False)
+    monkeypatch.setenv('TERM', 'xterm')
+    assert logger.should_do_markup(FileHandleTTY(True))
+
+
+def test_markup_detection_tty_no(monkeypatch):
+    monkeypatch.delenv('PY_COLORS', raising=False)
+    assert not logger.should_do_markup(FileHandleTTY(False))
```

Comment:
According to `pytest` practices, it is typical to use fixtures for such things.
You could do
```python
@pytest.fixture
def not_tty():
    return FileHandleTTY(False)

def test_markup_detection_tty_no(monkeypatch, not_tty):
```